libnavigation-ui/src/main/java/com/mapbox/navigation/ui/internal/route/MapboxRouteLayerProvider.kt

```diff
@@ -44,27 +44,22 @@ import com.mapbox.navigation.ui.internal.route.RouteConstants.WAYPOINT_PROPERTY_
 import com.mapbox.navigation.ui.internal.route.RouteConstants.WAYPOINT_SOURCE_ID
 import com.mapbox.navigation.ui.internal.utils.MapImageUtils
 import com.mapbox.navigation.ui.route.RouteStyleDescriptor
+import kotlin.reflect.KProperty1
 
 interface MapboxRouteLayerProvider : RouteLayerProvider {
     val routeStyleDescriptors: List<RouteStyleDescriptor>
 
-    fun getRouteLineColorExpressions(defaultColor: Int): List<Expression> {
+    fun getRouteLineColorExpressions(
+        defaultColor: Int,
+        routeColorProvider: KProperty1<RouteStyleDescriptor, Int>
```

Comment:
Do we really need `KProperty1`? `reflect` in general is scary. Also I'm wondering if that may introduce problems with Java compatibility as it's a Kotlin type 👀 at the ⚠️ https://kotlinlang.org/docs/reference/reflection.html#reflection

Reply:
This saves from writing a lot of duplicate code in this case and the class is internal now so nobody external to the team will be accessing it.

Reply:
Ohh but we are as most of the UI codebase is in Java 😬

Reply:
Only wondering about the implications of depending on `reflect`.